tests/test_str_utils.py

```diff
@@ -0,0 +1,21 @@
+from __future__ import annotations
+
+import pytest
+
+from qadence import Z, hamiltonian_factory, kron
+from qadence.blocks import AbstractBlock, MatrixBlock, KronBlock
+from qadence.utils import blocktree_to_mathematical_expression
+import torch
+
+
+@pytest.mark.parametrize(
```

Comment:
Cool, it seems that there is one more case that codedev complains about. If you can cover it that'd be awesome.